src/Linux/curl_easy.cpp

```diff
@@ -86,7 +86,11 @@ char const* curl_easy::error::what() const noexcept
 ///////////////////////////////////////////////////////////////////////////////
 // curl_easy implementation
 ///////////////////////////////////////////////////////////////////////////////
-std::unique_ptr<curl_easy> curl_easy::create(const std::string& url, const std::string* const p_body, unsigned long dwflag)
+std::unique_ptr<curl_easy> curl_easy::create(
+    const std::string& url,
+    const std::string* const p_body,
+    unsigned long dwflag,
+    bool fetchingFromLocalAgent)

```

Comment:
NOTE: Updating the code based the following feedback is optional.

It is usually better to separate low-level from high-level implementation details. This makes low-level code more reusable and context agnostic. In this case, `fetchingFromLocalAgent` is a high-level implementation detail but curl_easy may be used in many other contexts for which "fetching from agent" will not make sense. Maybe calling this `shouldTimeout` will keep it general, an even better approach (IMO) would be to make the caller pass the timeout so that this can be reused in any other places where different timeouts apply.

Reply:
that's a good point but since the scope is limited here, the current implementation should be okay.